inc/common.class.php

```diff
@@ -246,4 +246,159 @@ public static function isCurrentUser(PluginFlyvemdmAgent $item) {
+                  $defaultDate->setTimezone(new \DateTimeZone(date_default_timezone_get()));
+                  $default = $defaultDate->format("Y-m-d H:i:s");
+               } else if ($column['COLUMN_DEFAULT'] > '2038-01-19 03:14:07') {
+                  // Prevent default value to be out of range (greater to max possible value)
+                  $defaultDate = new \DateTime('2038-01-19 03:14:07', new \DateTimeZone('UTC'));
+                  $defaultDate->setTimezone(new \DateTimeZone(date_default_timezone_get()));
+                  $default = $defaultDate->format("Y-m-d H:i:s");
+               } else {
+                  $default = $column['COLUMN_DEFAULT'];
+               }
+            }
+
+            //build alter
+            $tablealter .= "\n\t MODIFY COLUMN ".$DB->quoteName($column['COLUMN_NAME'])." TIMESTAMP";
+            if ($nullable) {
+               $tablealter .= " NULL";
+            } else {
+               $tablealter .= " NOT NULL";
+            }
+            if ($default !== null) {
+               if ($default !== 'NULL') {
+                  $default = "'" . $DB->escape($default) . "'";
+               }
+               $tablealter .= " DEFAULT $default";
+            }
+            if ($column['COLUMN_COMMENT'] != '') {
+               $tablealter .= " COMMENT '".$DB->escape($column['COLUMN_COMMENT'])."'";
+            }
+            $tablealter .= ",";
+         }
+         $tablealter =  rtrim($tablealter, ",");
+
+         // apply alter to table
+         $query = "ALTER TABLE " . $DB->quoteName($table['TABLE_NAME']) . " " . $tablealter.";\n";
+
+         $migration->displayWarning(sprintf(__('Running %s'), $query));
+
+         $result = $DB->query($query);
+         if (false === $result) {
+            $message = sprintf(
+               __('Update of `%s` failed with message "(%s) %s".'),
+               $table['TABLE_NAME'],
+               $DB->errno(),
+               $DB->error()
+            );
+            $migration->displayWarning($message);
+         }
+      }
+   }
+
+   /**
+    * Determines the timezone to use
+    * @return String timezone
+    */
+   public static function guessTimezone() {
+      //At the first connection if the user has defined a timezone in these preferences
+      //GLPI set it in the $_SESSION
+      if (isset($_SESSION['glpi_tz'])) {
+         $zone = $_SESSION['glpi_tz'];
+      } else {
+         //If not set in $_SESSION try to load timezone from GLPI gerneral config
+         $conf_tz = ['value' => null];
+         if ($this->tableExists(Config::getTable())
+            && $this->fieldExists(Config::getTable(), 'value')
+         ) {
+            $conf_tz = $this->request([
+               'SELECT' => 'value',
+               'FROM'   => Config::getTable(),
+               'WHERE'  => [
+                  'context'   => 'core',
+                  'name'      => 'timezone'
+               ]
+            ])->next();
```

Comment:
Why not use 

```php
$config = Config::getConfigurationValues('core', 'timezone');
```

and an extra check with

```php
if (isset($config['timezone')) { ... }
```

Reply:
I simplified the code to find the timezone